ons_alpha/core/blocks/markup.py

```diff
@@ -32,23 +33,75 @@ class Meta:
         icon = "info-circle"
         template = "templates/components/streamfield/table_block.html"
 
+    def __init__(self, required=True, help_text=None, table_options=None, **kwargs):
+        if table_options is None:
+            table_options = {
+                "mergeCells": True,
+                "contextMenu": DEFAULT_TABLE_OPTIONS["contextMenu"] + ["---------", "mergeCells", "alignment"],
+            }
+
+        super().__init__(required=required, help_text=help_text, table_options=table_options, **kwargs)
+
+    def _to_ons_classname(self, classname: str) -> str:
+        """
+        Reference: https://handsontable.com/docs/javascript-data-grid/text-alignment/#horizontal-and-vertical-alignment
+        """
+        match classname:
+            case "htRight":
+                return "ons-u-ta-right"
+            case "htLeft":
+                return "ons-u-ta-left"
+            case "htCenter":
+                return "ons-u-ta-center"
+            case _:
+                return ""
+
+    def _get_rows(self, value, table_header, classnames):
+        trs = []
+        data = value["data"][1:] if table_header else value.get("data", [])
+        for row_idx, row in enumerate(data, 1 if table_header else 0):
+            tds = []
+            for cell_idx, cell in enumerate(row):
+                td = {"value": cell}
+                if classname := classnames.get((row_idx, cell_idx)):
+                    td["tdClasses"] = classname
+                tds.append(td)
+
+            trs.append({"tds": tds})
+        return trs
+
     def get_context(self, value, parent_context=None):
-        context = super().get_context(value, parent_context)  #
+        context = super().get_context(value, parent_context=parent_context)
+
+        classnames = {}
+        hidden = {}
+        spans = {}
+        if value.get("cell"):
+            for meta in value["cell"]:
+                if "className" in meta:
+                    classnames[(meta["row"], meta["col"])] = self._to_ons_classname(meta["className"])
+                if "hidden" in meta:
+                    hidden[(meta["row"], meta["col"])] = meta["hidden"]
+
+        if value.get("mergeCells"):
+            for merge in value["mergeCells"]:
+                spans[(merge["row"], merge["col"])] = {
+                    "rowspan": merge["rowspan"],
+                    "colspan": merge["colspan"],
+                }
```

Comment:
note: these are lifted from Wagtail core.
`hidden`/`span` are ready to plug in when the component would support them